src/creature.cpp

```diff
@@ -102,6 +103,24 @@ void Creature::reset()
     reset_bonuses();
     reset_stats();
 }
+
+void Creature::bleed( const tripoint &where, int intensity ) const
```

Comment:
> It should be blood splatter function's role to do the bleeding, critter should only provide blood type.

I started to think similarly when had to deal with gore and also found blood spreading in mapgen. So I tend to agree.

> Hallucinations can be handled by giving them no blood type.

Good idea. I think it should be handled in `monster::bloodType()`.

> EDIT: By "blood splatter function" I mean the cause of bleeding. The function in ranged.cpp and the similar one in vehicle.cpp. They determine the trajectory of blood, so they should handle the bleeding on too.

There are even more of them scattered over the code. IMHO there should me as few of such functions, as possible, and they shouldn't belong neither to ranged.cpp nor vehicle.cpp.